src/main/java/com/github/lunatrius/schematica/client/world/SchematicWorld.java

```diff
@@ -34,12 +35,45 @@
 public class SchematicWorld extends WorldClient {
     private static final WorldSettings WORLD_SETTINGS = new WorldSettings(0, GameType.CREATIVE, false, false, WorldType.FLAT);
 
+    public static enum LayerMode {
+        ALL(Names.Gui.Control.MODE_ALL) {
+            @Override
+            public boolean shouldUseLayer(final SchematicWorld world, final int layer) {
+                return true;
+            }
+        },
+        SINGLE_LAYER(Names.Gui.Control.MODE_LAYERS) {
+            @Override
+            public boolean shouldUseLayer(final SchematicWorld world, final int layer) {
+                return layer == world.renderingLayer;
+            }
+        },
+        ALL_BELOW(Names.Gui.Control.MODE_BELOW) {
+            @Override
+            public boolean shouldUseLayer(final SchematicWorld world, final int layer) {
+                return layer <= world.renderingLayer;
+            }
+        };
+        public abstract boolean shouldUseLayer(SchematicWorld world, int layer);
+
+        private LayerMode(String name) {
+            this.name = name;
+        }
+
+        public final String name;
+
+        public static LayerMode next(final LayerMode mode) {
+            LayerMode[] values = values();
+            return values[(mode.ordinal() + 1) % values.length];
+        }
+    }
+
     private ISchematic schematic;
 
     public final MBlockPos position = new MBlockPos();
-    public boolean isRendering;
```

Comment:
I needed to move the initialization from the constructor to here because otherwise `layerMode` would be `null` in `initializeTileEntity`, which causes problems.